src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.NativeAot.cs

```diff
@@ -188,30 +177,80 @@ private static unsafe IntPtr GetSymbol(IntPtr handle, string symbolName, bool th
             return ret;
         }
 
-        // TODO: copy the nice error logic from CoreCLR's LoadLibErrorTracker
-        // to get fine-grained error messages that take into account access denied, etc.
+        // Preserving good error info from DllImport-driven LoadLibrary is tricky because we keep loading from different places
+        // if earlier loads fail and those later loads obliterate error codes.
+        //
+        // This tracker object will keep track of the error code in accordance to priority:
+        //
+        //   low-priority:      unknown error code (should never happen)
+        //   medium-priority:   dll not found
+        //   high-priority:     dll found but error during loading
+        //
+        // We will overwrite the previous load's error code only if the new error code is higher priority.
         internal struct LoadLibErrorTracker
         {
-            internal const int ERROR_INVALID_PARAMETER = 0x57;
-            internal const int ERROR_MOD_NOT_FOUND = 126;
-            internal const int ERROR_BAD_EXE_FORMAT = 193;
-
+#if TARGET_WINDOWS
```

Comment:
Now that there is even more Windows/Unix-specific code, it would be nice to split it into separate .Windows/.Unix partial files to follow the coding conventions.